task-force-docs/osm/names.md

```diff
@@ -0,0 +1,194 @@
+zh_pinyin -> zh-Latn-pinyin
+zh_zhuyin -> zh-Bopo
+be-tarask -> be-Latn-tarask
+nan-POJ -> nan-Latn
+```
+
+### Overture Variants
+
+Overture only defines a few variants. At the time of this document's writing
+they are: `common`, `official`, `local`, and `short`.   The formal list is
+defined in the Overture schema.
+
+OSM has many variants which are mapped to one of the Overture variants:
+| OSM Variant | Overture Variant |
+| ----------- | ---------------- |
+| none | common |
+| official | official |
+| short | short |
+| loc | local |
+| int | alternate |
+| nat | alternate |
+| old | alternate |
+| ref | alternate |
+| reg | alternate |
+| alt | alternate |
+| nick | alternate |
+
+All other variants are ignored.
+
+
+## Populating the Overture Names Field
+
+Once all valid variants and language tags have been derived from OSM tags, populating
+the Overture names field is as follows:
+
+* primary
+`primary` is always the "common" variant with no language tag. This amounts to the bar `name=*` tag in OSM
+
+* common
+`common` is a map of the common variants where the keys are the BCP47 language tags and the values are their corresponding
+tag values. Because it is a map there can only be one entry for each language.
+
+* rules
+`rules` is an array of structs that captures all other variants. Each rule structure has the following fields:
+
+`language` - the converted language tag
+`variant` - the name variant
+`value` - the tag value
+`between` - used for linear-referencing of transportation segments. This is always null for non-transportation elements.
+`side` - used when one side of a transportation segment has a name. This is always null for non-transportation elements.
+
+Because `rules` is a list there are no restrictions on the number of variants or
+languages. It is possible to have multiple instances of the same variant and
+language combination. 
+
+
+
+
+## Example
+
+For example, the New York City label node (https://www.openstreetmap.org/node/61785451) has some of the following
+name tags:
+```
+name=New York
+name:br=Evrog Nevez
+name:el=Νέα Υόρκη
+name:es=Nueva York
+name:be-tarask=Нью-Ёрк
+old_name:es=Nueva Ámsterdam
+loc_name=Big Apple
+loc_name:es=La Gran Manzana
+official_name=City of New York
+...
+```
+
+Here is how this is converted into the corresponding Overture names.
+
+```
+{
+    primary: "New York",
+    common: {
+        "br": "Evrog Nevez",
+        "el": "Νέα Υόρκη",
+        "es": "Nueva York",
+        "be-Latn-tarask": "Нью-Ёрк"
+    },
+    rules: [
+        {
+            "value": "City of New York",
+            "variant": "official",
+            "language": null,
+            "between": null,
+            "side": null
+        },
+        {
+            "value": "Nueva Ámsterdam",
+            "variant": "alternate",
+            "language": "es",
+            "between": null,
+            "side": null
+        },
+        {
+            "value": "Big Apple",
+            "variant": "local",
```

Comment:
`local` doesn't exist in Overture currently. Can this also be `alternate`?

Reply:
Whoops! I actually thought for a while `local` was a variant. Just fixed it. Thanks!